core/xmipp_image_base.cpp

```diff
@@ -883,9 +912,12 @@ int ImageBase::_readBatch(const FileName &name, ImageFHandler* hFile, size_t sta
     if (ext_name.contains("spi") || ext_name.contains("xmp")  ||
         ext_name.contains("stk") || ext_name.contains("vol")) {
         err = readSPIDER(start_img, batch_size);
-    } else if (ext_name.contains("mrcs")||ext_name.contains("st")) { //mrc stack MUST go BEFORE plain MRC
+    } else if (ext_name.contains("mrcs") || ext_name.contains("st") ||
```

Comment:
duplicated code. Extract methods such as
`isMRCType(ext)` etc.
Don't forget to call that method in the write methods as well

Reply:
Done